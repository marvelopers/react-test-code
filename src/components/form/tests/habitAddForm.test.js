import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import HabitAddForm from "../habitAddForm";

describe("HabitAddForm", () => {
  it("renders - 스냅샷 테스트", () => {
    const component = renderer.create(<HabitAddForm onAdd={jest.fn()} />);
  });

  describe("Form submit", () => {
    let onAdd;
    let input;
    let button;
    beforeEach(() => {
      onAdd = jest.fn();
      render(<HabitAddForm onAdd={onAdd} />);

      input = screen.getByPlaceholderText("Habit");
      button = screen.getByText("Add");
    });

    it("calls onAdd when button is Clicked and valid habit is entered", () => {
      userEvent.type(input, "New Habit");
      userEvent.click(button);

      expect(onAdd).toHaveBeenCalledWith("New Habit");
    });

    it("does not call onAdd when the habit is empty", () => {
      userEvent.type(input, "");
      userEvent.click(button);

      expect(onAdd).toBeCalledTimes(0);
    });
  });
});
