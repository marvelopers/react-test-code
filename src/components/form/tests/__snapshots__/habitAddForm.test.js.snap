// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitAddForm renders - 스냅샷 테스트 1`] = `
<form
  className="add-form"
  onSubmit={[Function]}
>
  <input
    className="add-input"
    placeholder="Habit"
    type="text"
  />
  <button
    className="add-button"
  >
    Add
  </button>
</form>
`;
